### DISCUSSION TITLE
Segmentation example with 2D image

### DISCUSSION THREAD

**Discussion by kitu-datta** [Project-MONAI/MONAI#5216] [Q&A]
Hi,

I was trying to implement 2D segmentation , following the mentioned code as below
(https://github.com/Project-MONAI/tutorials/blob/master/2d_segmentation/torch/unet_training_array.py)
I see the 2D images and labels that are created have (128, 128) dimension, however I have dataset with images and labels of (512, 512). I trained the model and the mean dice is coming as: 1.8621, also the output image is blank. 

I am using the code as is mentioned.  I am not clear what I am missing here.
I am new to monai, would appreciate for a guidance on using MONAI U-Net Segmentation for 2D image with single label.

**Comment by Nic-Ma** [comment]
Hi @kitu-datta ,

Thanks for using MONAI in your projects.
I think there should be no difference between (128, 128) and (512, 512) when using the UNet, maybe some other difference in your dataset? For example: data with channel or not? multi-labels or multi-classes?
Could you please help paste your test program here for further debug?

Thanks.

**Reply by kitu-datta** [reply]
Hi @Nic-Ma,

Thank you for your response.

Initially I am converting RGB to grayscale image , 

#Block Image set
  for image in images:
      file_name = os.path.basename(image)
      fName, ext = os.path.splitext(file_name)      
      color_im = io.imread(image)
      gray_im = color.rgb2gray(color_im)
      io.imsave(os.path.join(BASE_PATH, f'Data/block_images/{fName}.png'), gray_im)

#Block Label set
  for seg in segs:
      file_name = os.path.basename(seg)
      fName, ext = os.path.splitext(file_name)     
      color_im = io.imread(seg)
      gray_im = color.rgb2gray(color_im)
      io.imsave(os.path.join(BASE_PATH, f'Data/block_labels/{fName}.png'), gray_im)

and keeping the rest of the code as is mentioned. Also to mention, I have single-class or single-label .

Thanks

**Comment by rijobro** [comment]
What do your images look like when you put them through the train transforms? e.g., `train_imtrans(os.path.join(BASE_PATH, f'Data/block_labels/{fName}.png')`

**Reply by sarbanid** [reply]
Hi @rijobro,

For RandSpatialCrop((96, 96), random_size=False), the label images through train_transforms show blank.
I tried using RandSpatialCrop((256, 256), random_size=False), the label images through train_transforms is  not blank, infact I could see ~(75-90)% of label. As the object on the image is not consistent in its position on every images. However, after training the U-Net model with torch.Size([10, 1, 256, 256]) for image and label respectively, I do not see output image with label , it still shows blank.

**Reply by rijobro** [reply]
It sounds like you shouldn't be cropping so much then. If 256x256 only leaves ~75% of the segmentation, then 96x96 is going to leave even less, maybe even none.

N.B.: You can also use `CenterSpatialCrop` to crop the centre of the image, rather than a random part, which might reduce the amount of the segmentation that gets cropped (assuming that the segmentation is towards the centre of the image).

**Reply by kitu-datta** [reply]
Thanks @rijobro ,

Using CenterSpatialCrop() in transform helped. Also, using ScaleIntensity() in label  the dice metric value is 0.96 now.

Thank you, this resolved my issue.

**Comment by witness97** [comment]
Hi @kitu-datta ,
Glad to see your discussion. I meet the same problems as you met before. I try to use the multi-classes images to do the training but the mean dice I get is always larger than 1. My image and label size are [512, 512, 3]. Would you please tell me how to solve the problem?
Thanks.

**Comment by SenhorUnk** [comment]


**Reply by rijobro** [reply]
Hi, please create a new discussion. Should be a MONAI Label Discussion: https://github.com/Project-MONAI/MONAILabel/discussions.
